### PR TITLE
Upgrade operator-sdk to 1.9.0

### PR DESCRIPTION
**Pull request by pliurh**
Upgrade project to go.kubebuilder.io/v3

### CODE REVIEW

File: test/util/cluster/cluster.go [hunk 1]
```diff
@@ -19,6 +19,28 @@ import (
 	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/pod"
 )
 
+func init() {
```
Comment (zshi-redhat) [inline test/util/cluster/cluster.go:22]: maybe we shall call the init from sriovv1 like this: https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/155/files#diff-fa9c480471555909f4af5c083bdc918e887e487b12ab7eb49cc27bf10757bb0cR53 ?

File: controllers/suite_test.go [hunk 2]
```diff
@@ -61,21 +61,20 @@ const (
...
-		// UseExistingCluster: func() *bool { b := true; return &b }(),
+		CRDDirectoryPaths:     []string{filepath.Join("config", "crd", "bases"), filepath.Join("test", "util", "crds")},
+		ErrorIfCRDPathMissing: true,
 	}
 
-	var err error
-	cfg, err = testEnv.Start()
-	Expect(err).ToNot(HaveOccurred())
-	Expect(cfg).ToNot(BeNil())
+	cfg, err := testEnv.Start()
```
Comment (zshi-redhat) [inline controllers/suite_test.go:75]: Why we change ToNot to NotTo? I see we have both in other places.

Comment (pliurh) [reply controllers/suite_test.go:75]: This was generated by kube-builder. Both do the same work. I don't think it matters.

**Comment by zshi-redhat** [comment]
Some CustomResourceDefinition still use v1beta1 api version:

```
./test/util/crds/machineconfiguration.openshift.io_machineconfigs_crd.yaml:apiVersion: apiextensions.k8s.io/v1beta1
./test/util/crds/machineconfiguration.openshift.io_machineconfigpools_crd.yaml:apiVersion: apiextensions.k8s.io/v1beta1
./test/util/crds/k8s.cni.cncf.io_networkattachmentdefinitions_crd.yaml:apiVersion: apiextensions.k8s.io/v1beta1
```

**Comment by pliurh** [comment]
> Some CustomResourceDefinition still use v1beta1 api version:
> 
> ```
> ./test/util/crds/machineconfiguration.openshift.io_machineconfigs_crd.yaml:apiVersion: apiextensions.k8s.io/v1beta1
> ./test/util/crds/machineconfiguration.openshift.io_machineconfigpools_crd.yaml:apiVersion: apiextensions.k8s.io/v1beta1
> ./test/util/crds/k8s.cni.cncf.io_networkattachmentdefinitions_crd.yaml:apiVersion: apiextensions.k8s.io/v1beta1
> ```

I'll fix this in another PR.

**Comment by zshi-redhat** [comment]
Are there any notes when running make generate/manifest/bundle etc after this change? 
For example, the operator-sdk version etc?

Shall we add the notes in the commit message?

**Comment by pliurh** [comment]
> Are there any notes when running make generate/manifest/bundle etc after this change?
> For example, the operator-sdk version etc?
> 
> Shall we add the notes in the commit message?

How about creat a HACKING.md file that describes the details in another PR?

File: test/util/cluster/cluster.go (hunk 1 above)
Comment (pliurh) [reply test/util/cluster/cluster.go]: Remove `init()`, as it has been solved by #155.

**Pull request merged**